benchmarks/no-op_stress_test.cpp

```diff
@@ -0,0 +1,110 @@
+    time[0] += std::chrono::duration<double>(end_alloc - begin_alloc).count()/ALLOCATIONS;
+
+    auto begin_dealloc = std::chrono::system_clock::now();
+    for (int h = 0; h < ALLOCATIONS; h++)
+      alloc.deallocate(allocations[h]);
+    auto end_dealloc = std::chrono::system_clock::now();
+    time[1] += std::chrono::duration<double>(end_dealloc - begin_dealloc).count()/ALLOCATIONS;
+  }
+
+  std::cout << "    SAME_ORDER:" << std::endl; 
+  std::cout << "    alloc: " << (time[0]/double(NUM_ITER)*CONVERT) << "(us)" << std::endl;
+  std::cout << "    dealloc: " << (time[1]/double(NUM_ITER)*CONVERT) << "(us)" << std::endl << std::endl;
+}
+
+void reverse_order(std::size_t size, umpire::Allocator alloc)
+{
+  double time[2] = {0.0, 0.0};
+  void* allocations[ALLOCATIONS];
+  for(int i = 0; i < NUM_ITER; i++) {
+    auto begin_alloc = std::chrono::system_clock::now();
+    for (int j = 0; j < ALLOCATIONS; j++)
+      allocations[j] = alloc.allocate(size);
+    auto end_alloc = std::chrono::system_clock::now();
+    time[0] += std::chrono::duration<double>(end_alloc - begin_alloc).count()/ALLOCATIONS;
+
+    auto begin_dealloc = std::chrono::system_clock::now();
+    for (int h = (ALLOCATIONS-1); h >=0; h--)
+      alloc.deallocate(allocations[h]);
+    auto end_dealloc = std::chrono::system_clock::now();
+    time[1] += std::chrono::duration<double>(end_dealloc - begin_dealloc).count()/ALLOCATIONS;
+  }
+
+  std::cout << "    REVERSE_ORDER:" << std::endl; 
+  std::cout << "    alloc: " << (time[0]/double(NUM_ITER)*CONVERT) << "(us)" << std::endl;
+  std::cout << "    dealloc: " << (time[1]/double(NUM_ITER)*CONVERT) << "(us)" << std::endl << std::endl;
+}
+
+void shuffle(std::size_t size, umpire::Allocator alloc)
+{
+  std::mt19937 gen(ALLOCATIONS);
+  double time[2] = {0.0, 0.0};
+  void* allocations[ALLOCATIONS];
+
+  for(int i = 0; i < NUM_ITER; i++) {
+    auto begin_alloc = std::chrono::system_clock::now();
+    for (int j = 0; j < ALLOCATIONS; j++)
+      allocations[j] = alloc.allocate(size);
+    auto end_alloc = std::chrono::system_clock::now();
+    time[0] += std::chrono::duration<double>(end_alloc - begin_alloc).count()/ALLOCATIONS;
+
+    std::shuffle(&allocations[0], &allocations[ALLOCATIONS], gen);
+    auto begin_dealloc = std::chrono::system_clock::now();
+    for (int h = 0; h < ALLOCATIONS; h++)
+      alloc.deallocate(allocations[h]);
+    auto end_dealloc = std::chrono::system_clock::now();
+    time[1] += std::chrono::duration<double>(end_dealloc - begin_dealloc).count()/ALLOCATIONS;
+  }
+
+  std::cout << "    SHUFFLE:" << std::endl; 
+  std::cout << "    alloc: " << (time[0]/double(NUM_ITER)*CONVERT) << "(us)" << std::endl;
+  std::cout << "    dealloc: " << (time[1]/double(NUM_ITER)*CONVERT) << "(us)" << std::endl << std::endl;
+}
+
+int main(int, char**) {
+  std::mt19937 gen(ALLOCATIONS);
+  std::uniform_int_distribution<std::size_t> dist(64, 4096);
+  std::size_t size = dist(gen);
```

Comment:
Can you change this to use a fixed size of 4096. I think we want to keep as many variables fixed as possible to try and isolate what this is benchmarking.

Reply:
Done!